docs/modelserving/observability/grafana_dashboards.md

```diff
@@ -0,0 +1,39 @@
+
+## KServe TorchServe Latency Dashboard
+
+A template dashboard for [KServe TorchServe Latency](https://grafana.com/grafana/dashboards/18026-kserve-torchserve-latency/) contains an inference latency graph which plots the [rate](https://prometheus.io/docs/prometheus/latest/querying/functions/#rate) of the TorchServe metric `ts_inference_latency_microseconds` in milliseconds. A second graph plots the rate of TorchServe's internal queue latency metric `ts_queue_latency_microseconds` in milliseconds. A third graph plots the total requests to the TorchServe Inference Service. 
+
+## KServe Triton Latency Dashboard 
+
+A template dashboard for [KServe Triton Latency](https://grafana.com/grafana/dashboards/18027-kserve-triton-latency/) contains five latency graphs with the rate of Triton's input (preprocess), infer (predict), output (postprocess), internal queue and total latency metrics plotted in milliseconds. Triton outputs metrics on GPU usage as well, and the template plots a gauge of the percentage of GPU memory usage in bytes.
+
+## Debugging 
+
+With these Grafana dashboards set up, debug latency issues with the following steps
+
+First, (if in serverless mode) start with the Knative HTTP Dashboard to check if there is a queueing delay with queue-proxy
+
+- [x] compare the gateway latency percentile metrics with your target SLO  
+- [x] check the observed concurrency metrics to see if your service overloaded with a high number of inflight requests, indicating the service is over capacity and is unable to keep up with the number of requests 
+- [x] check the GPU/CPU memory metrics to see if the service is close to its limits - if your service has a high number of inflight requests/high CPU/GPU usage, then a possible solution is to add more resources/replicas
+
+Next, take a look at the appropriate serving runtime dashboard to see if there is a bottleneck in the code
+
+- [x] check the latencies for pre/post-process, predict, explain - are latencies higher than expected at any one step? If so, you may need to make changes/adjustments for this step (note: TorchServe does not currently expose this level of observability at the moment, only an inference latency graph which encompasses the steps together)
+- [x] check the queue latency metrics (TorchServe and Triton) - if requests are stuck in the queue, the model is not able to keep up with the number of requests, consider adding more resources/replicas
+- [x] (Triton) check the GPU utilization metrics to see if your service is at capacity and you need more GPU resources
```

Comment:
TorchServe has GPU utilization metrics as well

Reply:
I don't see them output whereas I do see triton's 🤔

Reply:
```
curl -v localhost:8082/metrics
*   Trying 127.0.0.1:8082...
* Connected to localhost (127.0.0.1) port 8082 (#0)
> GET /metrics HTTP/1.1
> Host: localhost:8082
> User-Agent: curl/7.79.1
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< content-type: text/plain; version=0.0.4; charset=utf-8
< x-request-id: 10f2de41-6c13-4dd6-9b21-557c5720b189
< Pragma: no-cache
< Cache-Control: no-cache; no-store, must-revalidate, private
< Expires: Thu, 01 Jan 1970 00:00:00 UTC
< content-length: 509
< connection: keep-alive
<
# HELP ts_inference_latency_microseconds Cumulative inference duration in microseconds
# TYPE ts_inference_latency_microseconds counter
# HELP ts_inference_requests_total Total number of inference requests.
# TYPE ts_inference_requests_total counter
ts_inference_requests_total{uuid="c2d182e7-4b86-4d4a-866b-d877943ad6f0",model_name="text-classification",model_version="default",} 4.0
# HELP ts_queue_latency_microseconds Cumulative queue duration in microseconds
# TYPE ts_queue_latency_microseconds counter
* Connection #0 to host localhost left intact
```

Reply:
You do not see them as your testing service is not deployed on GPU